app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -713,28 +723,35 @@ void OpenXRInputSource::EmulateControllerFromHand(device::RenderMode renderMode,
     }
 #endif
 
-    if (renderMode == device::RenderMode::StandAlone)
-        pointerTransform.TranslateInPlace(kAverageHeight);
+    device::CapabilityFlags flags = device::Orientation | device::Position;
+    if (pointerMode == DeviceDelegate::PointerMode::TRACKED_POINTER) {
+        if (renderMode == device::RenderMode::StandAlone)
+            pointerTransform.TranslateInPlace(kAverageHeight);
 
+        // TODO: removing this flag will allow us to remove the ifdef. Keeping it just to limit
+        // the scope of the changes.
+        flags = device::GripSpacePosition;
 #if CHROMIUM
-    // Blink WebXR uses the grip space instead of the local space to position the
-    // controller. Since controller's position is the same as the origin of the
-    // grip space, we just set the transform matrix to the identity.
-    // Then we need to correct the beam transform matrix to maintain origin and
-    // direction when we are not in immersive mode.
-    delegate.SetTransform(mIndex, vrb::Matrix::Identity());
-    delegate.SetBeamTransform(mIndex, pointerTransform);
+        // Blink WebXR uses the grip space instead of the local space to position the
+        // controller. Since controller's position is the same as the origin of the
+        // grip space, we just set the transform matrix to the identity.
+        // Then we need to correct the beam transform matrix to maintain origin and
+        // direction when we are not in immersive mode.
+        delegate.SetTransform(mIndex, vrb::Matrix::Identity());
+        delegate.SetBeamTransform(mIndex, pointerTransform);
 #else
-    delegate.SetTransform(mIndex, pointerTransform);
-    delegate.SetBeamTransform(mIndex, vrb::Matrix::Identity());
+        delegate.SetTransform(mIndex, pointerTransform);
+        delegate.SetBeamTransform(mIndex, vrb::Matrix::Identity());
 #endif
-    delegate.SetImmersiveBeamTransform(mIndex, pointerTransform);
+        delegate.SetImmersiveBeamTransform(mIndex, pointerTransform);
+    } else if (pointerMode == DeviceDelegate::PointerMode::TRACKED_EYE) {
```

Comment:
nit: Better to do an plain "else" and and assert inside.

Reply:
Sure. I'll change it.